### PR TITLE
Fix minmax result definition

### PR DESCRIPTION
**Pull request by MdrditArthur**
M functions min & max should return an undefined M value if both arguments are undefined, as it is the behaviour of the old tax system compiler.

### CODE REVIEW

File: src/mlang/backend_compilers/bir_to_dgfip_c.ml [hunk 1]
```diff
@@ -819,13 +819,6 @@ let generate_rovs_files (dgfip_flags : Dgfip_options.flags) (program : program)
 #define add_erreur(a,b,c) add_erreur(b,c)
 #endif
 
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)
```
Comment (denismerigoux) [inline src/mlang/backend_compilers/bir_to_dgfip_c.ml:822]: Where is the replacement for those?

Comment (MdrditArthur) [reply src/mlang/backend_compilers/bir_to_dgfip_c.ml:822]: _fmin/_fmax are replaced in the generated code by min/max. 
min/max are already defined as macros (equivalent to the _fmin/_fmax when C < 1999 was used) in the static const.h file, included by the function `generate_header` line 583.

Comment (denismerigoux) [reply src/mlang/backend_compilers/bir_to_dgfip_c.ml:822]: Ok they're here 

https://github.com/MLanguage/mlang/blob/510b5d457f2a6c42c729c70fdfae7a70e986a60e/examples/dgfip_c/const.h#L109-L110

File: src/mlang/optimizing_ir/partial_evaluation.ml [hunk 2]
```diff
@@ -618,27 +618,13 @@ let rec partially_evaluate_expr (ctx : partial_ev_ctx) (p : Mir.program)
...
-                    (fun a d ->
-                      if Pos.unmark a = Mir.Literal Undefined || d = Undefined
-                      then Pos.same_pos_as (Mir.Literal (Float 0.)) a
-                      else a)
-                    new_args new_ds
-                in
-                let new_e =
-                  Pos.same_pos_as (Mir.FunctionCall (func, new_args)) e
-                in
-                (Pos.unmark new_e, Float)
```
Comment (Keryan-dev) [inline src/mlang/optimizing_ir/partial_evaluation.ml:641]: It's a shame not to reduce those when we can.

Please comment that we should do it at some point.

Comment (Keryan-dev) [reply src/mlang/optimizing_ir/partial_evaluation.ml:641]: Ah, too late. Well then, later I guess.

Comment (MdrditArthur) [reply src/mlang/optimizing_ir/partial_evaluation.ml:641]: Would you like an issue on the topic ?

Comment (Keryan-dev) [reply src/mlang/optimizing_ir/partial_evaluation.ml:641]: #194 :)

Comment (Keryan-dev) [reply src/mlang/optimizing_ir/partial_evaluation.ml:641]: To be clearer, it does not patch multimax. But I don't think it's worth trying to considering how the pass works and how little chance there is to actually simplify it statically.

**Pull request merged**